crates/core/plugin_sm/src/plugin.rs

```diff
@@ -1,60 +1,64 @@
 use async_trait::async_trait;
 use csv::ReaderBuilder;
 use download::Downloader;
-use logged_command::LoggedCommand;
 use regex::Regex;
 use reqwest::Identity;
 use serde::Deserialize;
 use std::error::Error;
 use std::path::Path;
 use std::path::PathBuf;
 use std::process::Output;
-use tedge_api::*;
+use tedge_api::CommandLog;
+use tedge_api::DownloadInfo;
+use tedge_api::LoggedCommand;
+use tedge_api::SoftwareError;
+use tedge_api::SoftwareModule;
+use tedge_api::SoftwareModuleUpdate;
+use tedge_api::SoftwareType;
+use tedge_api::DEFAULT;
 use tedge_config::SudoCommandBuilder;
-use tokio::fs::File;
 use tokio::io::AsyncWriteExt;
-use tokio::io::BufWriter;
 use tracing::error;
 
 #[async_trait]
 pub trait Plugin {
-    async fn prepare(&self, logger: &mut BufWriter<File>) -> Result<(), SoftwareError>;
+    async fn prepare(&self, command_log: Option<&mut CommandLog>) -> Result<(), SoftwareError>;
```

Comment:
To cover the case where the `logPath` is not provided in the payload of the command (since it is optional) when an external component has overridden the workflows, but still using the `builtin` actions for the actual installation.